karapace/schema_reader.py

```diff
@@ -337,29 +342,35 @@ def _handle_msg_schema(self, key: dict, value: Optional[dict]) -> None:
             try:
                 ujson.loads(schema_str)
             except ValueError:
-                self._log.error("Invalid json: %s", value["schema"])
+                self._log.error("Schema is not invalid JSON")
+                self._log.log(TRACE, "Invalid json: %s", value["schema"])
                 return
 
-        typed_schema = TypedSchema(schema_type=schema_type_parsed, schema_str=schema_str)
-        self._log.debug("Got typed schema %r", typed_schema)
-
         if schema_subject not in self.subjects:
             self._log.info("Adding first version of subject: %r with no schemas", schema_subject)
             self.subjects[schema_subject] = {"schemas": {}}
 
         subjects_schemas = self.subjects[schema_subject]["schemas"]
 
-        if schema_version in subjects_schemas:
-            self._log.info("Updating entry for subject: %r, value: %r", schema_subject, value)
-        else:
-            self._log.info("Adding new version of subject: %r, value: %r", schema_subject, value)
-
-        subjects_schemas[schema_version] = {
+        typed_schema = TypedSchema(
+            schema_type=schema_type_parsed,
+            schema_str=schema_str,
+        )
+        schema = {
             "schema": typed_schema,
             "version": schema_version,
             "id": schema_id,
             "deleted": schema_deleted,
         }
+
+        if schema_version in subjects_schemas:
```

Comment:
Particular reason why the location of this logging has been changed?

Reply:
because of what you said https://github.com/aiven/karapace/pull/384#discussion_r842390928